### ISSUE TITLE
Crash on fclose

### ISSUE THREAD

**Issue by jcarroll-mediafly** [marmelroy/Zip#99]
Looks like there are times when `filePointer` is nil and doesn't go into the while loop. Without further investigating im not sure why. In this case, however, we should not be closing the pointer as it will cause a crash.

```
var filePointer: UnsafeMutablePointer<FILE>?
            filePointer = fopen(fullPath, "wb")
            while filePointer != nil {
                let readBytes = unzReadCurrentFile(zip, &buffer, bufferSize)
                if readBytes > 0 {
                    fwrite(buffer, Int(readBytes), 1, filePointer)
                }
                else {
                    break
                }
            }

                fclose(filePointer)
```

**Comment by jcarroll-mediafly** [comment]
Hm, turns out subfolders in the archieve aren't archiving for me.

Using the convenient Zip.unzip function

**Comment by AvdLee** [comment]
Fixed with #128
